### PR TITLE
add docker-compose file

### PR DESCRIPTION
**Pull request by NL-TCH**
i build and tested this on my laptop, can you guys verify if this works on raspberry pi?

### CODE REVIEW

**Comment by jrcichra** [comment]
Hmm, not sure what isn't working yet. This is me using `CMD [ "/bin/bash", "-c", "/home/firewall-rules.sh && /lib/systemd/systemd" ]`, which works with `docker run`:

```
pi@justinpi:~/git/raspap-docker $ docker-compose logs -f
Attaching to raspap
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:36 2024
raspap    | *filter
raspap    | :INPUT ACCEPT [1271965:152826984]
raspap    | :FORWARD DROP [59:5664]
raspap    | :OUTPUT ACCEPT [802907:2092012500]
raspap    | :DOCKER - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-1 - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-2 - [0:0]
raspap    | :DOCKER-USER - [0:0]
raspap    | -A FORWARD -j DOCKER-USER
raspap    | -A FORWARD -j DOCKER-ISOLATION-STAGE-1
raspap    | -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -o docker0 -j DOCKER
raspap    | -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
raspap    | -A FORWARD -i docker0 -o docker0 -j ACCEPT
raspap    | -A FORWARD -i eth0 -o wlan0 -m state --state RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -i wlan0 -o eth0 -j ACCEPT
raspap    | -A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
raspap    | -A DOCKER-ISOLATION-STAGE-1 -j RETURN
raspap    | -A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
raspap    | -A DOCKER-ISOLATION-STAGE-2 -j RETURN
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:36 2024
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:36 2024
raspap    | *nat
raspap    | :PREROUTING ACCEPT [177913:24009102]
raspap    | :INPUT ACCEPT [169513:22274570]
raspap    | :OUTPUT ACCEPT [2519:184949]
raspap    | :POSTROUTING ACCEPT [0:0]
raspap    | :DOCKER - [0:0]
raspap    | -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
raspap    | -A POSTROUTING -j MASQUERADE
raspap    | -A POSTROUTING -s 192.168.50.0/24 ! -d 192.168.50.0/24 -j MASQUERADE
raspap    | -A POSTROUTING -o eth0 -j MASQUERADE
raspap    | -A DOCKER -i docker0 -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:36 2024
raspap    | Explicit --user argument required to run as user manager.
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:37 2024
raspap    | *filter
raspap    | :INPUT ACCEPT [1271966:152827036]
raspap    | :FORWARD DROP [59:5664]
raspap    | :OUTPUT ACCEPT [802908:2092012692]
raspap    | :DOCKER - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-1 - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-2 - [0:0]
raspap    | :DOCKER-USER - [0:0]
raspap    | -A FORWARD -j DOCKER-USER
raspap    | -A FORWARD -j DOCKER-ISOLATION-STAGE-1
raspap    | -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -o docker0 -j DOCKER
raspap    | -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
raspap    | -A FORWARD -i docker0 -o docker0 -j ACCEPT
raspap    | -A FORWARD -i eth0 -o wlan0 -m state --state RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -i wlan0 -o eth0 -j ACCEPT
raspap    | -A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
raspap    | -A DOCKER-ISOLATION-STAGE-1 -j RETURN
raspap    | -A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
raspap    | -A DOCKER-ISOLATION-STAGE-2 -j RETURN
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:37 2024
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:37 2024
raspap    | *nat
raspap    | :PREROUTING ACCEPT [177913:24009102]
raspap    | :INPUT ACCEPT [169513:22274570]
raspap    | :OUTPUT ACCEPT [2519:184949]
raspap    | :POSTROUTING ACCEPT [0:0]
raspap    | :DOCKER - [0:0]
raspap    | -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
raspap    | -A POSTROUTING -j MASQUERADE
raspap    | -A POSTROUTING -s 192.168.50.0/24 ! -d 192.168.50.0/24 -j MASQUERADE
raspap    | -A POSTROUTING -o eth0 -j MASQUERADE
raspap    | -A DOCKER -i docker0 -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:37 2024
raspap    | Explicit --user argument required to run as user manager.
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:37 2024
raspap    | *filter
raspap    | :INPUT ACCEPT [1271973:152827719]
raspap    | :FORWARD DROP [59:5664]
raspap    | :OUTPUT ACCEPT [802914:2092028690]
raspap    | :DOCKER - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-1 - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-2 - [0:0]
raspap    | :DOCKER-USER - [0:0]
raspap    | -A FORWARD -j DOCKER-USER
raspap    | -A FORWARD -j DOCKER-ISOLATION-STAGE-1
raspap    | -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -o docker0 -j DOCKER
raspap    | -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
raspap    | -A FORWARD -i docker0 -o docker0 -j ACCEPT
raspap    | -A FORWARD -i eth0 -o wlan0 -m state --state RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -i wlan0 -o eth0 -j ACCEPT
raspap    | -A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
raspap    | -A DOCKER-ISOLATION-STAGE-1 -j RETURN
raspap    | -A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
raspap    | -A DOCKER-ISOLATION-STAGE-2 -j RETURN
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:37 2024
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:37 2024
raspap    | *nat
raspap    | :PREROUTING ACCEPT [177913:24009102]
raspap    | :INPUT ACCEPT [169513:22274570]
raspap    | :OUTPUT ACCEPT [2519:184949]
raspap    | :POSTROUTING ACCEPT [0:0]
raspap    | :DOCKER - [0:0]
raspap    | -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
raspap    | -A POSTROUTING -j MASQUERADE
raspap    | -A POSTROUTING -s 192.168.50.0/24 ! -d 192.168.50.0/24 -j MASQUERADE
raspap    | -A POSTROUTING -o eth0 -j MASQUERADE
raspap    | -A DOCKER -i docker0 -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:37 2024
raspap    | Explicit --user argument required to run as user manager.
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:38 2024
raspap    | *filter
raspap    | :INPUT ACCEPT [1271984:152828471]
raspap    | :FORWARD DROP [59:5664]
raspap    | :OUTPUT ACCEPT [802920:2092029626]
raspap    | :DOCKER - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-1 - [0:0]
raspap    | :DOCKER-ISOLATION-STAGE-2 - [0:0]
raspap    | :DOCKER-USER - [0:0]
raspap    | -A FORWARD -j DOCKER-USER
raspap    | -A FORWARD -j DOCKER-ISOLATION-STAGE-1
raspap    | -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -o docker0 -j DOCKER
raspap    | -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
raspap    | -A FORWARD -i docker0 -o docker0 -j ACCEPT
raspap    | -A FORWARD -i eth0 -o wlan0 -m state --state RELATED,ESTABLISHED -j ACCEPT
raspap    | -A FORWARD -i wlan0 -o eth0 -j ACCEPT
raspap    | -A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
raspap    | -A DOCKER-ISOLATION-STAGE-1 -j RETURN
raspap    | -A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
raspap    | -A DOCKER-ISOLATION-STAGE-2 -j RETURN
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -i src_if -o dst_if -j ACCEPT
raspap    | -A DOCKER-USER -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:38 2024
raspap    | # Generated by iptables-save v1.8.9 (nf_tables) on Wed Jan 10 02:48:38 2024
raspap    | *nat
raspap    | :PREROUTING ACCEPT [177913:24009102]
raspap    | :INPUT ACCEPT [169513:22274570]
raspap    | :OUTPUT ACCEPT [2519:184949]
raspap    | :POSTROUTING ACCEPT [0:0]
raspap    | :DOCKER - [0:0]
raspap    | -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
raspap    | -A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
raspap    | -A POSTROUTING -j MASQUERADE
raspap    | -A POSTROUTING -s 192.168.50.0/24 ! -d 192.168.50.0/24 -j MASQUERADE
raspap    | -A POSTROUTING -o eth0 -j MASQUERADE
raspap    | -A DOCKER -i docker0 -j RETURN
raspap    | COMMIT
raspap    | # Completed on Wed Jan 10 02:48:38 2024
raspap    | Explicit --user argument required to run as user manager.
raspap exited with code 1
pi@justinpi:~/git/raspap-docker $ 

```

I tried adding `cgroup: host` as documented here: https://docs.docker.com/compose/compose-file/05-services/#cgroup but by bookworm raspberry pi + docker-compose from apt doesn't know about it:

```
pi@justinpi:~/git/raspap-docker $ docker-compose up -d
ERROR: The Compose file './docker-compose.yaml' is invalid because:
Unsupported config option for services.raspap: 'cgroup'
```
pi@justinpi:~/git/raspap-docker $ docker-compose --version
docker-compose version 1.29.2, build unknown
```

```
pi@justinpi:~/git/raspap-docker $ cat docker-compose.yaml 
services:
  raspap:
    container_name: raspap
    #image: ghcr.io/raspap/raspap-docker:latest
    build: .
    privileged: true
    network_mode: host
    pid: host
    cgroup: host
    cap_add:
      - SYS_ADMIN
    volumes:
      - /sys/fs/cgroup:/sys/fs/cgroup:rw
    restart: unless-stopped 
```

**Comment by NL-TCH** [comment]
and without cgroups: host, does it work? isn't pid a replacement of cgroup ?

**Comment by NL-TCH** [comment]
i'll look at this one in the weekend, this might be related to the `version` in the yaml,

note to self: https://docs.docker.com/compose/compose-file/compose-file-v3/

**Comment by NL-TCH** [comment]
other note to self: https://github.com/docker/compose/issues/8167#issuecomment-1790773617

i'll update the dockerfile and add a readme section in a few days

**Comment by NL-TCH** [comment]
fixed it! can you guys verify?

**Comment by billz** [comment]
@NL-TCH will give it a test and provide feedback, thanks!

**Comment by NL-TCH** [comment]
@jrcichra i think you missed the added readme section :)
https://github.com/docker/compose/issues/8167#issuecomment-1790773617

TLDR: use `docker compose` instead of `docker-compose`

**Comment by NL-TCH** [comment]
wow yeah just recommend ppl to install via get.docker.com because the repo version of compose is from 2021 (https://docs.docker.com/compose/release-notes/#1292)

**Comment by billz** [comment]
yep, good find... you beat me to it :) agreed we should state something to this effect in the readme
